### ISSUE TITLE
Fatal Error during update, Access to undeclared static property: Sabre\VObject\Component::$classMap in /var/www/owncloud/apps/contacts/appinfo/app.php

### ISSUE THREAD

**Issue by enoch85** [owncloud/core#17452]
Steps to reproduce

1. Upgrade from 8.0.4 to 8.1 using occ command:

```
apt-get update
aptitude full-upgrade -y
sudo -u www-data php /var/www/owncloud/occ upgrade

sudo -u www-data php /var/www/owncloud/occ app:enable mail
sudo -u www-data php /var/www/owncloud/occ app:enable calendar
sudo -u www-data php /var/www/owncloud/occ app:enable contacts
sudo -u www-data php /var/www/owncloud/occ app:enable documents
```

First run:

```
Current status: 0 updates [-22].
ownCloud or one of the apps require upgrade - only a limited number of commands                                                             are available
Turned on maintenance mode
Checked database schema update
Checked database schema update for apps
Updated database
Disabled incompatible app: calendar
Disabled 3rd-party app: calendar
Disabled incompatible app: contacts
Disabled 3rd-party app: contacts
Disabled 3rd-party app: documents
Disabled 3rd-party app: mail
Disabled 3rd-party app: music
Disabled 3rd-party app: tasks
Updating <activity> ...
Updated <activity> to 2.0.1
Updating <files_sharing> ...
Updated <files_sharing> to 0.6.2
Updating <files_trashbin> ...
Updated <files_trashbin> to 0.6.3
Updating <files_versions> ...
Updated <files_versions> to 1.0.6
Update 3rd-party app: calendar
Update successful
Turned off maintenance mode
mail enabled
calendar enabled
ownCloud or one of the apps require upgrade - only a limited number of commands                                                             are available
contacts enabled
ownCloud or one of the apps require upgrade - only a limited number of commands                                                             are available
documents enabled
ownCloud or one of the apps require upgrade - only a limited number of commands                                                             are available
tasks enabled
root@en0ch:~# reboot
```

Second run:

```
ownCloud is already latest version
mail is already enabled
calendar is already enabled
contacts enabled
PHP Fatal error:  Access to undeclared static property: Sabre\VObject\Component:                                                            :$classMap in /var/www/owncloud/apps/contacts/appinfo/app.php on line 15
PHP Fatal error:  Access to undeclared static property: Sabre\VObject\Component:                                                            :$classMap in /var/www/owncloud/apps/contacts/appinfo/app.php on line 15
find: `/var/www/owncloud/data/daniel@en0ch.se/cache/ownCloud_www.en0ch.se-flat.v                                                            mdk-chunking-1428473208-1169': No such file or directory
root@en0ch:~# cd /var/www/owncloud/
root@en0ch:/var/www/owncloud# sudo -u www-data php occ maintenance:mode --on
PHP Fatal error:  Access to undeclared static property: Sabre\VObject\Component::$classMap in /var/www/owncloud/apps/contacts/appinfo/app.php on line 15
root@en0ch:/var/www/owncloud# sudo -u www-data php occ app:disable contacts
PHP Fatal error:  Access to undeclared static property: Sabre\VObject\Component::$classMap in /var/www/owncloud/apps/contacts/appinfo/app.php on line 15
```

2. Now I got WSOD and can't access ownCloud at all so I figured I will delete the contacts app from the system, and so I did. 

3. Disable contacts

```
root@en0ch:/var/www/owncloud# sudo -u www-data php occ app:disable contacts
contacts disabled
root@en0ch:/var/www/owncloud#
```
Now it works again, but shouldn't be like this.

### Expected behavior
ownCloud should upgrade to 8.1 without any issues

### Actual Behavior
I get WSOD because some error with the contacts app, and have to delete the contacts app to make it run again.

### Server configuration
**Operating system**:
Ubuntu Server 14.04.2
**Web server:**

**Database:**
Server version: 5.5.43-0ubuntu0.14.04.1 (Ubuntu)

**PHP version:**
PHP 5.5.9-1ubuntu4.11 (cli) (built: Jul  2 2015 15:23:08)
Copyright (c) 1997-2014 The PHP Group
Zend Engine v2.5.0, Copyright (c) 1998-2014 Zend Technologies
    with Zend OPcache v7.0.3, Copyright (c) 1999-2014, by Zend Technologies

**ownCloud version:** (see ownCloud admin page)
8.1

**Updated from an older ownCloud or fresh install:**
Updated from 8.0.4

**List of activated apps:**

```
Enabled:
 - activity: 2.0.1
 - calendar: 0.7.1
 - documents: 0.9.0
 - files: 1.1.9
 - files_locking:
 - files_pdfviewer: 0.7
 - files_sharing: 0.6.2
 - files_texteditor: 0.4
 - files_trashbin: 0.6.3
 - files_versions: 1.0.6
 - files_videoviewer: 0.1.3
 - firstrunwizard: 1.1
 - gallery: 0.6.0
 - mail: 0.1.3
 - tasks: 0.6.0
 - templateeditor: 0.1
Disabled:
 - encryption
 - external
 - files_external
 - music
 - provisioning_api
 - search_lucene
 - user_external
 - user_ldap
 - user_webdavauth

```

**The content of config/config.php:**

```
<?php
$CONFIG = array (
  'instanceid' => 'oc106ccbdf7c',
  'passwordsalt' => 'SECRET',
  'datadirectory' => '/var/www/owncloud/data',
  'dbtype' => 'mysql',
  'version' => '8.1.0.8',
  'dbname' => 'owncloud',
  'dbhost' => 'en0ch.se',
  'dbtableprefix' => 'oc_',
  'dbuser' => 'oc_enoch',
  'dbpassword' => 'SECRET',
  'installed' => true,
  'maxZipInputSize' => 0,
  'allowZipDownload' => true,
  'session_lifetime' => 1800,
  'forcessl' => true,
  'updatechecker' => true,
  'loglevel' => 2,
  'theme' => '',
  'maintenance' => false,
  'trusted_domains' =>
  array (
    0 => 'en0ch.se',
    1 => 'www.en0ch.se',
  ),
  'preview_libreoffice_path' => '/usr/bin/libreoffice',
  'enable_previews' => true,
  'app.mail.server-side-cache.enabled' => true,
  'app.mail.imaplog.enabled' => true,
  'logtimezone' => 'Europe/Stockholm',
```

**Are you using external storage, if yes which one:** local/smb/sftp/...
No
**Are you using encryption:** yes/no
No
**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...
No

#### ownCloud log (data/owncloud.log)
```
{"reqId":"daad5e7547fe84cd3e9ef3495d443533","remoteAddr":"79.138.133.115","app":"PHP","message":"Cannot modify header information - headers already sent by (output started at \/var\/www\/owncloud\/lib\/private\/image.php:249) at \/var\/www\/owncloud\/lib\/private\/appframework\/app.php#100","level":3,"time":"2015-07-07T17:10:50+02:00"}
{"reqId":"daad5e7547fe84cd3e9ef3495d443533","remoteAddr":"79.138.133.115","app":"PHP","message":"Cannot modify header information - headers already sent by (output started at \/var\/www\/owncloud\/lib\/private\/image.php:249) at \/var\/www\/owncloud\/lib\/private\/appframework\/app.php#104","level":3,"time":"2015-07-07T17:10:50+02:00"}
{"reqId":"daad5e7547fe84cd3e9ef3495d443533","remoteAddr":"79.138.133.115","app":"PHP","message":"Cannot modify header information - headers already sent by (output started at \/var\/www\/owncloud\/lib\/private\/image.php:249) at \/var\/www\/owncloud\/lib\/private\/appframework\/app.php#104","level":3,"time":"2015-07-07T17:10:50+02:00"}
{"reqId":"daad5e7547fe84cd3e9ef3495d443533","remoteAddr":"79.138.133.115","app":"PHP","message":"Cannot modify header information - headers already sent by (output started at \/var\/www\/owncloud\/lib\/private\/image.php:249) at \/var\/www\/owncloud\/lib\/private\/appframework\/app.php#104","level":3,"time":"2015-07-07T17:10:50+02:00"}
{"reqId":"e6683f2af114ffd45bae84b3787b50d6","remoteAddr":"79.138.133.115","app":"PHP","message":"Cannot modify header information - headers already sent by (output started at \/var\/www\/owncloud\/lib\/private\/image.php:249) at \/var\/www\/owncloud\/lib\/private\/appframework\/app.php#100","level":3,"time":"2015-07-07T17:10:51+02:00"}
{"reqId":"e6683f2af114ffd45bae84b3787b50d6","remoteAddr":"79.138.133.115","app":"PHP","message":"Cannot modify header information - headers already sent by (output started at \/var\/www\/owncloud\/lib\/private\/image.php:249) at \/var\/www\/owncloud\/lib\/private\/appframework\/app.php#104","level":3,"time":"2015-07-07T17:10:51+02:00"}
{"reqId":"e6683f2af114ffd45bae84b3787b50d6","remoteAddr":"79.138.133.115","app":"PHP","message":"Cannot modify header information - headers already sent by (output started at \/var\/www\/owncloud\/lib\/private\/image.php:249) at \/var\/www\/owncloud\/lib\/private\/appframework\/app.php#104","level":3,"time":"2015-07-07T17:10:51+02:00"}
{"reqId":"e6683f2af114ffd45bae84b3787b50d6","remoteAddr":"79.138.133.115","app":"PHP","message":"Cannot modify header information - headers already sent by (output started at \/var\/www\/owncloud\/lib\/private\/image.php:249) at \/var\/www\/owncloud\/lib\/private\/appframework\/app.php#104","level":3,"time":"2015-07-07T17:10:51+02:00"}
{"reqId":"433dfa10ab45089cc87de582d9f53fdd","remoteAddr":"79.138.133.115","app":"PHP","message":"Cannot modify header information - headers already sent by (output started at \/var\/www\/owncloud\/lib\/private\/image.php:249) at \/var\/www\/owncloud\/lib\/private\/appframework\/app.php#100","level":3,"time":"2015-07-07T17:10:52+02:00"}
{"reqId":"433dfa10ab45089cc87de582d9f53fdd","remoteAddr":"79.138.133.115","app":"PHP","message":"Cannot modify header information - headers already sent by (output started at \/var\/www\/owncloud\/lib\/private\/image.php:249) at \/var\/www\/owncloud\/lib\/private\/appframework\/app.php#104","level":3,"time":"2015-07-07T17:10:52+02:00"}
{"reqId":"433dfa10ab45089cc87de582d9f53fdd","remoteAddr":"79.138.133.115","app":"PHP","message":"Cannot modify header information - headers already sent by (output started at \/var\/www\/owncloud\/lib\/private\/image.php:249) at \/var\/www\/owncloud\/lib\/private\/appframework\/app.php#104","level":3,"time":"2015-07-07T17:10:52+02:00"}
{"reqId":"433dfa10ab45089cc87de582d9f53fdd","remoteAddr":"79.138.133.115","app":"PHP","message":"Cannot modify header information - headers already sent by (output started at \/var\/www\/owncloud\/lib\/private\/image.php:249) at \/var\/www\/owncloud\/lib\/private\/appframework\/app.php#104","level":3,"time":"2015-07-07T17:10:52+02:00"}
{"reqId":"8b4c5d0ec626283379d98c6a504e3f65","remoteAddr":"79.138.133.115","app":"PHP","message":"Cannot modify header information - headers already sent by (output started at \/var\/www\/owncloud\/lib\/private\/image.php:249) at \/var\/www\/owncloud\/lib\/private\/appframework\/app.php#100","level":3,"time":"2015-07-07T17:10:53+02:00"}
{"reqId":"8b4c5d0ec626283379d98c6a504e3f65","remoteAddr":"79.138.133.115","app":"PHP","message":"Cannot modify header information - headers already sent by (output started at \/var\/www\/owncloud\/lib\/private\/image.php:249) at \/var\/www\/owncloud\/lib\/private\/appframework\/app.php#104","level":3,"time":"2015-07-07T17:10:53+02:00"}
{"reqId":"8b4c5d0ec626283379d98c6a504e3f65","remoteAddr":"79.138.133.115","app":"PHP","message":"Cannot modify header information - headers already sent by (output started at \/var\/www\/owncloud\/lib\/private\/image.php:249) at \/var\/www\/owncloud\/lib\/private\/appframework\/app.php#104","level":3,"time":"2015-07-07T17:10:53+02:00"}
{"reqId":"8b4c5d0ec626283379d98c6a504e3f65","remoteAddr":"79.138.133.115","app":"PHP","message":"Cannot modify header information - headers already sent by (output started at \/var\/www\/owncloud\/lib\/private\/image.php:249) at \/var\/www\/owncloud\/lib\/private\/appframework\/app.php#104","level":3,"time":"2015-07-07T17:10:53+02:00"}
{"reqId":"a70420dbc830f5146db80175eefe89ef","remoteAddr":"79.138.133.115","app":"PHP","message":"Cannot modify header information - headers already sent by (output started at \/var\/www\/owncloud\/lib\/private\/image.php:249) at \/var\/www\/owncloud\/lib\/private\/appframework\/app.php#100","level":3,"time":"2015-07-07T17:10:55+02:00"}
{"reqId":"a70420dbc830f5146db80175eefe89ef","remoteAddr":"79.138.133.115","app":"PHP","message":"Cannot modify header information - headers already sent by (output started at \/var\/www\/owncloud\/lib\/private\/image.php:249) at \/var\/www\/owncloud\/lib\/private\/appframework\/app.php#104","level":3,"time":"2015-07-07T17:10:55+02:00"}
{"reqId":"a70420dbc830f5146db80175eefe89ef","remoteAddr":"79.138.133.115","app":"PHP","message":"Cannot modify header information - headers already sent by (output started at \/var\/www\/owncloud\/lib\/private\/image.php:249) at \/var\/www\/owncloud\/lib\/private\/appframework\/app.php#104","level":3,"time":"2015-07-07T17:10:55+02:00"}
{"reqId":"a70420dbc830f5146db80175eefe89ef","remoteAddr":"79.138.133.115","app":"PHP","message":"Cannot modify header information - headers already sent by (output started at \/var\/www\/owncloud\/lib\/private\/image.php:249) at \/var\/www\/owncloud\/lib\/private\/appframework\/app.php#104","level":3,"time":"2015-07-07T17:10:55+02:00"}
{"reqId":"bf44a98573dfcb96d99622f25573423d","remoteAddr":"79.138.133.115","app":"PHP","message":"Cannot modify header information - headers already sent by (output started at \/var\/www\/owncloud\/lib\/private\/image.php:249) at \/var\/www\/owncloud\/lib\/private\/appframework\/app.php#100","level":3,"time":"2015-07-07T17:10:56+02:00"}
{"reqId":"bf44a98573dfcb96d99622f25573423d","remoteAddr":"79.138.133.115","app":"PHP","message":"Cannot modify header information - headers already sent by (output started at \/var\/www\/owncloud\/lib\/private\/image.php:249) at \/var\/www\/owncloud\/lib\/private\/appframework\/app.php#104","level":3,"time":"2015-07-07T17:10:56+02:00"}
```

**Comment by RealRancor** [comment]
Very likely a dup of: https://github.com/owncloud/core/issues/16314

**Comment by enoch85** [comment]
@RealRancor Maybe, but this is the contacts app. Calendar has no issues anymore, at least for me.

**Comment by PVince81** [comment]
I think the contacts app wasn't published yet for 8.1, so you probably got the 8.0 version which isn't compatible

**Comment by PVince81** [comment]
@DeepDiver1975 ?

**Comment by PVince81** [comment]
@enoch85 the CLI upgrade doesn't auto-download the latest code for third party apps AFAIK.
You have to do enable the app from the web UI to make that happen.

Hmmm, but I think it would make sense to have CLI app enabling also auto-download/update/install the app ?

**Comment by RealRancor** [comment]
Contacts 0.4.0.0 works here just fine with oC 8.1

**Comment by PVince81** [comment]
Ok, then it means it's "./occ app:enable" that doesn't auto-download the latest code of the apps.
I'll confirm this.

**Comment by PVince81** [comment]
Just tried it locally and I was wrong.
`./occ upgrade` does try and update an app.

I got the line "Update 3rd-party app: calendar" that would download the latest code.

Now the question is why @enoch85 didn't get such line for the contact apps ?
@enoch85 can you check the "apps/contacts/appinfo/version" file and tell us what version it is ?

**Comment by PVince81** [comment]
Let me try again with the contacts app. Maybe the latest version wasn't on the app store yet when you tried.

**Comment by PVince81** [comment]
### Steps

1. Check out "v8.0.5" tag from git
1. Setup OC in the web UI
1. Enable contacts app from the "Productivity" section, it should be version 0.3.0.19
1. Check out "v8.1.0" tag from git
1. Run `./occ upgrade` in the CLI

```
Turned on maintenance mode
Checked database schema update
Checked database schema update for apps
Updated database
Disabled 3rd-party app: admin_migrate
Disabled incompatible app: contacts
Disabled 3rd-party app: contacts
Disabled 3rd-party app: files_odfviewer
Disabled 3rd-party app: user_migrate
Update 3rd-party app: contacts
GuzzleHttp\Exception\RequestException: cURL error 60: SSL certificate problem: unable to get local issuer certificate
Update failed
Maintenance mode is kept active
```
It almost worked for me, the SSL error is due to https://github.com/owncloud/core/issues/17446 and is something with my environment.

Nevertheless, this shows that the upgrade process did attempt to update the code.

@enoch85 can you just try again exactly the same as you did before ? I suspect that the contact apps was shortly published to the correct version.

**Comment by enoch85** [comment]
@PVince81 Like i wrote in the issue, I deleted the contacts app and downloaded it again from the apps menu. It works now, but just wanted to inform you regarding this bug. 

Could this be closed then?

**Comment by enoch85** [comment]
Confirmed. Contacts app is working. Tried on another server.

```
Turned on maintenance mode
Checked database schema update
Checked database schema update for apps
Updated database
Disabled 3rd-party app: calendar
Disabled 3rd-party app: contacts
Disabled 3rd-party app: documents
Disabled 3rd-party app: mail
Updating <activity> ...
Updated <activity> to 2.0.1
Updating <files_sharing> ...
Updated <files_sharing> to 0.6.2
Updating <files_trashbin> ...
Updated <files_trashbin> to 0.6.3
Updating <files_versions> ...
Updated <files_versions> to 1.0.6
Update 3rd-party app: calendar
Update 3rd-party app: contacts
Update successful
Turned off maintenance mode
mail enabled
calendar enabled
contacts enabled
ownCloud or one of the apps require upgrade - only a limited number of commands are available
documents enabled
ownCloud or one of the apps require upgrade - only a limited number of commands are available
```
Still an issue that the occ command don't update the apps properly and I have to press "start update" again when I enter the domain, but that's already known.

**Comment by PVince81** [comment]
@enoch85 thanks for the update.

**Comment by GrahamCobb** [comment]
I don't see how this can be closed.  The upgrade is completely broken.

I just did apt-get dist-upgrade, then restarted apache2, logged in to my account (using the web access, allowed the upgrade to go ahead, went to the apps (still using the web GUI), re-enabled calendar, clicked on the app upgrade and I am now screwed.  I just a blank screen and the same error in the log.

I even tried "sudo -u www-data php /var/www/owncloud/occ app:disable calendar" and just get the same error.

How do I recover? Why doesn't the upgrade work? What has happened to my data?

Will the same thing happen when I try to re-enable Contacts?

**Comment by RealRancor** [comment]
Delete the /apss/calendar folder and enable the correct version from tge GUI later

**Comment by sava345** [comment]
Broken for both contacts and calendar
- owncloud displays blank page, status code is 500, server error log shows missing this and that
move both /var/www/owncloud/apps/calendar and /var/www/owncloud/apps/contacts away
download 8.1 versions and unzip/untar to the place of the previous folders
https://apps.owncloud.com/content/show.php/Contacts?content=168708
https://apps.owncloud.com/content/show.php/Calendar?content=168707
load owncloud homepage, goes into upgrade mode, then both apps are enabled.

**Comment by milgner** [comment]
Just came across the same problem, but it worked after I updated the Contacts app using the 8.1 download from the given link and ran `occ upgrade` once more.

**Comment by mschaaf** [comment]
Still an issue for me too.

**Comment by RealRancor** [comment]
@mschaaf Just install the version of the app which is compatible with oC 8.1.
